pandas/core/apply.py

```diff
@@ -323,55 +359,28 @@ def agg_or_apply_list_like(
         if getattr(obj, "axis", 0) == 1:
             raise NotImplementedError("axis other than 0 is not supported")
 
+        selected_obj = obj
         if not isinstance(obj, SelectionMixin):
             # i.e. obj is Series or DataFrame
             selected_obj = obj
         elif obj._selected_obj.ndim == 1:
+            assert False
             # For SeriesGroupBy this matches _obj_with_exclusions
             selected_obj = obj._selected_obj
         else:
+            assert False
             selected_obj = obj._obj_with_exclusions
```

Comment:
`selected_obj` is always the same as `obj`
 here?

Reply:
Correct; I cleaned this up.